amazon/src/main/java/amazon/robot/AwaitingJobState.java

```diff
@@ -0,0 +1,77 @@
+package amazon.robot;
+
+import amazon.*;
+import simulator.*;
+import simulator.utils.*;
+
+class AwaitingJobState implements RobotState {
+  private final Robot context;
+
+  public AwaitingJobState(Robot context) {
+    this.context = context;
+  }
+
+  @Override
+  public void tick(Warehouse warehouse, int currentTickCount) throws Exception {
+    // If at Charging Pod, charge, otherwise move towards it.
+    if (this.context.getLocation().equals(this.context.chargingPod.getLocation()))
+      this.context.charge();
```

Comment:
Wouldn't it make more sense that the charging pod charged the robot whenever it was standing on top of it? That way, the robot is only concerned about getting there.